sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala

```diff
@@ -69,9 +75,19 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
     }
   }
 
-  protected def initProc: (OutputStream, Process, InputStream, CircularBuffer) = {
-    val cmd = List("/bin/bash", "-c", script)
```

Comment:
To resolve the issue you mentioned, can we just simply the current working directory into `PATH`?

Reply:
Can't work in this way.

Reply:
Why is that? You can call it via `/bin/bash`:

```
$ echo "#! /usr/bin/python\nprint('hello')" > test.py
$ chmod 755 test.py
$ PATH=PATH$:`pwd` /bin/bash -c "test.py"
hello
```

Reply:
> Why is that? You can call it via `/bin/bash`:
> 
> ```
> $ echo "#! /usr/bin/python\nprint('hello')" > test.py
> $ chmod 755 test.py
> $ PATH=$PATH:`pwd` /bin/bash -c "test.py"
> hello
> ```

Make a mistake, this way can make it work, When I test, the path env not changed.  The problem is how can I update PATH env properties in program?

Reply:
You're creating another process to invoke bash. You can set the environment in the process builder.

Reply:
> You're creating another process to invoke bash. You can set the environment in the process builder.

== Just put value to the environment can work.
So for handle my problem, put Spark root dir to Path can make it work. Do you support to add a command wrapper?
It will be more flexible  but may change current compatibility with previous usage.

Reply:
Does hive support such configuration too? It doesn't look particularly necessary to me.

Reply:
> Does hive support such configuration too? It doesn't look particularly necessary to me.

Yea, Hive support such configuration.  wait for other committer's review comment too?

Reply:
Yeah, I don't feel strongly. Why don't you just fix the issue you face first without the configuration first?

The new configuration looks orthogonal with the issue you faced anyway.

Reply:
> Yeah, I don't feel strongly. Why don't you just fix the issue you face first without the configuration first?
> 
> The new configuration looks orthogonal with the issue you faced anyway.

Because I didn't figured out why `/bin/bash -c` in the spark root dir can't run the program `xx.py`.
So I check the code about hive and with hive's code we can support all case then make this pr.
After you say add Spark root dir  to `PATH` I make it clear.... we treat `xx.py` as a program, we need to find it in `PATH`.

Reply:
> Does hive support such configuration too? It doesn't look particularly necessary to me.
>> Yea, Hive support such configuration. wait for other committer's review comment too?
>> gentle ping @cloud-fan @maropu

The @HyukjinKwon idea looks better and I think it's easier for users to use this new feature without any configuration.

Reply:
> The @HyukjinKwon idea looks better and I think it's easier for users to use this new feature without any configuration.

Yea

Reply:
Updated.